### PR TITLE
a-o-i: Separate install and scaleup workflows

### PR DESCRIPTION
**Pull request by smunilla**
In order to simplify the quick installer logic, this moves
the scaleup workflow into its own subcommand.

Fixes: Bug 1339621

### CODE REVIEW

**Comment by smunilla** [comment]
Unit tests that were checking the old scaleup workflow are obviously breaking. Working on fixing/revising those as needed.

**Comment by smunilla** [comment]
aos-ci-test

**Comment by tbielawa** [comment]
I set up a 2 node cluster locally which looks like this

> Hosts:
> - m01.example.com
>   - OpenShift master
>   - OpenShift node (Unscheduled)
>   - Etcd (Embedded)
>   - Storage
> - n01.example.com
>   - OpenShift node (Dedicated)
> 
> Total OpenShift masters: 1
> Total OpenShift nodes: 2

When I started to run the scaleup workflow I had to pause and read carefully. What I saw looked like I had just ran the quick installer `install` wf and was about to install the environment again:

```
*** Installation Summary ***

Hosts:
- m01.example.com
  - OpenShift master
  - OpenShift node (Unscheduled)
  - Etcd (Embedded)
  - Storage
- n01.example.com
  - OpenShift node (Dedicated)

Total OpenShift masters: 1
Total OpenShift nodes: 2

NOTE: Add a total of 3 or more masters to perform an HA installation.


This tool will add nodes to the installed environment listed.

Are you ready to continue? [y/N]:
```

There's only one line of text in there that mentions this is a different operation, "This tool will add nodes".

For me personally this screen was misleading and required more thought than I expected. *Specifically*, the **Installation Summary** header made me think I had ran the `atomic-openshift-installer install` workflow again and skipped to the last page.

For a scaleup workflow, maybe the first step can be something other than printing the current Installation Summary with a brief message.

For example, when I run the `install` workflow I am presented with a moderately sized wall-of-text with a header welcoming me to the **install** process. It's right there on the first line of output:

> Welcome to the OpenShift Enterprise 3 installation.

Just spit balling here, maybe something like the installer first page (but modified to fit the scale up context) is possible for the scaleup 'welcome' page? I like the welcome banner, and with some word changing I think it would be great for the scale up workflow too. Explain the requirements and what will happen afterwards ("When the process completes....").


After I pressed 'y' to continue I scrolled up in my backlog and saw what looks like a skipped page?

> *** New Node Configuration ***
> 
> Add new nodes here

Then I was brought to the **Host Configuration** page. The wording makes me feel like I am beginning a new installation from scratch.

> You must now specify the hosts that will compose your OpenShift cluster.

Maybe that should say something about "You must now specify the new hosts you will be adding to your existing OCP cluster"?

After I add my new host in the next prompt and confirm I have no other new hosts to add I am presented with the **Installation** summary again followed by the message "Gathering information from hosts". It might help users if we gave them more visual feedback here indicating which of the nodes in the installation summary are new. Example below with **[NEW]** on the line:

> Hosts:
> - m01.example.com
>   - OpenShift master
>   - OpenShift node (Unscheduled)
>   - Etcd (Embedded)
>   - Storage
> - n01.example.com
>   - OpenShift node (Dedicated)
> - n02.example.com [NEW]
>   - OpenShift node (Dedicated)

After I confirm the installation details "Do the above facts look correct? " the installation begins like normal except it's for just the one new host. That all good and well.

After installation I can see the new node is now in my cluster

```
[root@m01 ~]# oc get nodes
NAME              STATUS                     AGE
m01.example.com   Ready,SchedulingDisabled   24m
n01.example.com   Ready                      24m
n02.example.com   Ready                      1m
```

So in summary:

Pros: The new workflow functions correctly. And the changes to the old install workflow to direct me to the scaleup workflow are great too.

Cons: I think that some of the process is not clearly differentiated enough from the existing install workflow. Visual queues could be added to help users identify new items. Some section headers could be reworded. The first page of the scaleup workflow should be more like the install workflow banner.

**Comment by tbielawa** [comment]
@smunilla one more thing I forgot to mention, `./utils/docs/man/man1/atomic-openshift-installer.1.asciidoc.in` didn't get any :heart:

**Comment by smunilla** [comment]
@tbielawa Updated a bit based on your feedback!

File: utils/docs/man/man1/atomic-openshift-installer.1.asciidoc.in [hunk 1]
```diff
@@ -122,6 +123,11 @@ Upgrade to the latest major version. For example, if you are running
 version **3.2** then this could upgrade you to **3.3**.
 
 
+UPGRADE
```
Comment (tbielawa) [inline utils/docs/man/man1/atomic-openshift-installer.1.asciidoc.in]: I think that section title needs to change.

Comment (tbielawa) [inline utils/docs/man/man1/atomic-openshift-installer.1.asciidoc.in]: Also, make sure you re-run `make docs` from the `utils` directory and commit the changed `.1` file.

**Comment by smunilla** [comment]
aos-ci-test

File: utils/src/ooinstall/cli_installer.py [hunk 2]
```diff
@@ -1031,62 +1046,92 @@ def install(ctx, force, gen_inventory):
...
-    click.echo("Wrote Ansible inventory: %s" % inventory_file)
-    click.echo()
 
-    if gen_inventory:
-        sys.exit(0)
+@click.command()
+@click.option('--gen-inventory', is_flag=True, default=False,
+              help="Generate an Ansible inventory file and exit.")
+@click.pass_context
+def scaleup(ctx, gen_inventory):
```
Comment (tbielawa) [inline utils/src/ooinstall/cli_installer.py]: I'm Tim Blablabla and I approved this pull request.
